### PR TITLE
[common] Introduce Configuration and related utilities

### PR DESCRIPTION
**Pull request by PatrickRen**
This pull request introduces `Configuration` and its related utilities such as `ConfigOption`, and migrate current usage of Flink `Configuration` to CDC's own one in `flink-cdc-composer` and `flink-cdc-cli`.

Please note that most code blocks are borrowed from Flink, and some deprecated and redundant methods are dropped.

### CODE REVIEW

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/configuration/ConfigOptions.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (GOODBOY008) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/configuration/ConfigOptions.java:2]: License header should be `Ververica`?

**Comment by leonardBang** [comment]
@PatrickRen could you rebase your PR to latest master ?

File: flink-cdc-common/src/main/java/com/ververica/cdc/common/configuration/description/Description.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.common.configuration.description;
+
+import org.apache.flink.annotation.PublicEvolving;
```
Comment (leonardBang) [inline flink-cdc-common/src/main/java/com/ververica/cdc/common/configuration/description/Description.java:19]: We can use built-in annotation now?

**Review by leonardBang — APPROVED** [review]
Thanks @PatrickRen for the update, LGTM, I only left one comment, feel free to merge it once resolved it.

**Pull request merged**